### PR TITLE
Added css to hide recommended posts from print post view.

### LINKED ISSUE #13005: Related posts block: fix styling when printing the page
When _Related posts_ block is on the page and the page is printed, it looks awful:

image

TwentyNineteen theme.

Note that this might depend on the theme's styles and could potentially be improved directly in TwentyNineteen as well.

via p1HpG7-7d3-p2

### PR DESCRIPTION
**Pull request by spsiddarthan**
Fixes https://github.com/Automattic/jetpack/issues/16205
Fixes #13005

Jetpack should automatically hide extraneous elements like Likes, Sharing, and Related posts in print media. These are interactive elements and don't make sense in the context of a printed page.

In this PR, we are going to hide the "related posts" section in the "print" view. 

### Screenshots
Consider the below image, this is how related posts look as of now. 

Screenshot 2021-12-03 at 4 14 11 PM

If I were to print it: 
Screenshot 2021-12-07 at 8 02 59 PM

This PR introduces the changes to make the print view look like this

Screenshot 2021-12-07 at 8 03 37 PM

#### Changes proposed in this Pull Request:

* Added a CSS media query that hides the CSS class `.jp-relatedposts` in the print view. of the file  `projects/plugins/jetpack/modules/related-posts/related-posts.css` 

I add to add the `!important` to the css query for this class to override other rules being set before. Another option I considered was to introduce a new css parent container class, but I thought it added unnecessary overhead. 

Let me know if you have other ideas besides the `!important` solution which I have incorporated. 

Screenshot 2021-12-03 at 4 02 10 PM

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
 - Go to Jetpack->Settings->Traffic.
 - Enable related posts. 
 - You can either manually create posts or use some other means to create your posts. I used the `FakePress` plugin to create test data and applied the same tag on the posts that I wanted to be shown as related to each other. 
 - Go to any of the posts' page and observe the related posts section. 
 -  Now, print the page and in the print view, ensure that the related posts section doesn't show up. 
 - To test Dotcom changes, use a public WordPress.com site and sandbox it.

### CODE REVIEW

File: projects/plugins/jetpack/modules/related-posts/related-posts.css [hunk 1]
```diff
@@ -315,3 +315,13 @@
 		margin-right: 0;
 	}
 }
+
+/*
+ * To hide the realted post section in the print view of a post
+ */
```
Comment (danielpost) [inline projects/plugins/jetpack/modules/related-posts/related-posts.css:321]:
```suggestion
/*
 * Hide the related post section in the print view of a post
 */
```

**Review by danielpost — DISMISSED** [review]
Looks good to me! 👍🏼

**Review by jeherve — CHANGES_REQUESTED** [review]
You'll need to bump the version of the stylesheet so the cache is busted on update:
https://github.com/Automattic/jetpack/blob/edcc988c4e422fd3609ab00c80d2c5cb57b1f95b/projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php#L8

File: projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php [hunk 2]
```diff
@@ -5,7 +5,7 @@
 use Automattic\Jetpack\Sync\Settings;
 
 class Jetpack_RelatedPosts {
-	const VERSION   = '20210930';
+	const VERSION   = '20210931';
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php:8]: You can use today's date instead:

```suggestion
	const VERSION   = '20211209';
```

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php (hunk 2 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php]: @jeherve, done. Thank you so much. 

Q: In the future, if I just update JavaScript files, but not CSS,  I wouldn't have to bump the version, I reckon?

Comment (jeherve) [reply projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php]: It mostly depends on how the files are enqueued. When enqueuing a file in WordPress, you usually use the `wp_enqueue_*` functions. They allow you to specify a version number that's appended the resource's path as a query string, acting as a cache buster:

Screenshot 2021-12-09 at 14 05 31

In most Jetpack files, we use a constant to define the version, `JETPACK__VERSION`. That constant is updated every time we release a new version of Jetpack, so we don't have to worry about refreshing it, it's done with each new release.
Here is how Jetpack Likes's stylesheet is loaded, for example:
https://github.com/Automattic/jetpack/blob/4d6d490fb8f8f62a947917f080c33c95c4c8b2a6/projects/plugins/jetpack/modules/likes.php#L156

Related posts, however, use their own value for historical reasons:
https://github.com/Automattic/jetpack/blob/edcc988c4e422fd3609ab00c80d2c5cb57b1f95b/projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php#L1704

This is why we have to update it here.

**Review by jeherve — APPROVED** [review]

File: projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php (hunk 2 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php]: Thank you for the detailed explanation. That makes sense.

**Comment by jeherve** [comment]
r236832-wpcom

**Pull request merged**